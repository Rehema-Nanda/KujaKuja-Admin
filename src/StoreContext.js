// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
