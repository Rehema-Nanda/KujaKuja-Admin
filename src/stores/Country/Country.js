import {
    decorate, computed, observable, comparer,
} from "mobx";
import lineToPolygon from "@turf/line-to-polygon";

class Country {
    store = null;
    id = null;
    enabled = null;
    name = null;
    isoTwoLetterCode = null;
    geojson = null;
    lat = null;
    lng = null;
    createdAt = null;
    updatedAt = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            enabled: this.enabled,
            name: this.name,
            iso_two_letter_code: this.isoTwoLetterCode,
            geojson: JSON.stringify(this.geojson),
            lat: this.lat,
            lng: this.lng,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
        };
    }

    updateFromJson(json) {
        this.enabled = json.enabled;
        this.name = json.name;
        this.isoTwoLetterCode = json.iso_two_letter_code;
        this.geojson = json.geojson;
        this.lat = json.lat;
        this.lng = json.lng;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;

        this.fixGeoJson();
    }

    fixGeoJson() {
        // TODO: https://trello.com/c/wxv8WY0m/426-frontend-and-admin-backend-are-both-doing-linestring-to-polygon-conversion-but-differently-json-parsing
        if (this.geojson && this.geojson.geometry.type === "LineString") {
            this.geojson = lineToPolygon(this.geojson);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }
        await this.store.apiService.deleteCountry(this.id);
        await this.store.removeCountry(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateCountry(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createCountry(json);
            this.id = responseData.id;
            await this.store.addCountry(this);
        }
    }
}
decorate(Country, {
    lat: observable,
    lng: observable,
    asJson: computed({ equals: comparer.shallow }),
});

export default Country;
