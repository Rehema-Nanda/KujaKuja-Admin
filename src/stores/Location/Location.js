import {
    decorate, computed, observable, comparer,
} from "mobx";
import lineToPolygon from "@turf/line-to-polygon";
import circle from "@turf/circle";
import _ from "lodash";

class Location {
    store = null;
    id = null;
    name = null;
    geojson = null;
    lat = null;
    lng = null;
    createdAt = null;
    updatedAt = null;
    countryId = null;
    countryName = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            name: this.name,
            geojson: JSON.stringify(this.geojson),
            lat: this.lat,
            lng: this.lng,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
            country_id: this.countryId,
        };
    }

    updateFromJson(json) {
        this.name = json.name;
        this.geojson = json.geojson;
        this.lat = json.lat;
        this.lng = json.lng;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;
        this.countryId = json.country_id;
        this.countryName = json.country_name;

        this.fixGeoJson();
    }

    fixGeoJson() {
        // TODO: https://trello.com/c/wxv8WY0m/426-frontend-and-admin-backend-are-both-doing-linestring-to-polygon-conversion-but-differently-json-parsing
        if (this.geojson && typeof this.geojson === "string") {
            this.geojson = JSON.parse(this.geojson);
        }
        if (this.geojson && this.geojson.geometry.type === "LineString") {
            this.geojson = lineToPolygon(this.geojson);
        }

        // TODO: https://trello.com/c/9pBBU3hU/962-some-locations-are-missing-geometry
        // if no geometry, draw a 5km radius circle around coords
        if (this.geojson && _.isEmpty(this.geojson.geometry)) {
            const locationCoords = [this.lng, this.lat];
            const radius = 5;
            const options = { units: "kilometers" };
            this.geojson = circle(locationCoords, radius, options);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }
        await this.store.apiService.deleteLocation(this.id);
        await this.store.removeLocation(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateLocation(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createLocation(json);
            this.id = responseData.id;
            await this.store.addLocation(this);
        }
    }
}

decorate(Location, {
    lat: observable,
    lng: observable,
    asJson: computed({ equals: comparer.shallow }),
});

export default Location;
