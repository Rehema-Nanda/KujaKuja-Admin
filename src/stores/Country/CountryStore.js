import {
    flow, decorate, observable, computed,
} from "mobx";
import Country from "./Country";
import CountryApiService from "./CountryApiService";
import AppConfig from "../../AppConfig";

class CountryStore {
    countries = [];
    count = 0;
    enabledCountries = [];
    isLoading = false;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new CountryApiService(AppConfig.API_URL);

        this.loadCountries();
        this.loadEnabledCountries();
    }

    get countriesSortedByName() {
        return this.countries.slice().sort((a, b) => {
            if (a.name < b.name) {
                return -1;
            }
            else if (a.name > b.name) {
                return 1;
            }
            return 0;
        });
    }

    loadCountries = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchAllCountries(config);
            this.countries.replace(countriesJson.data.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.count = countriesJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadEnabledCountries = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchEnabledCountries();
            this.enabledCountries.replace(countriesJson.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load enabled countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateCountryFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const countriesJson = yield this.apiService.fetchCountryById(id);
        let country = this.countries.find((c) => {
            return c.id === id;
        });

        if (countriesJson.length === 0 && country) {
            // country no longer exists on the server, remove it locally
            yield this.removeCountry(country);
            country = null;
        }
        if (countriesJson.length > 0) {
            if (country) {
                country.updateFromJson(countriesJson[0]);
            }
            else {
                country = new Country(this, countriesJson[0]);
                this.countries.push(country);
            }
        }
        this.isLoading = false;
        return country;
    });

    removeCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.splice(this.countries.indexOf(country), 1);
    });

    addCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.push(country);
    });
}
decorate(CountryStore, {
    countries: observable,
    enabledCountries: observable,
    countriesSortedByName: computed,
    isLoading: observable,
});

export default CountryStore;
