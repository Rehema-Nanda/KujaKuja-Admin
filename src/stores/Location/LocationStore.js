import {
    flow, decorate, observable, computed,
} from "mobx";
import Location from "./Location";
import LocationApiService from "./LocationApiService";
import AppConfig from "../../AppConfig";

class LocationStore {
    locations = [];
    locationsSyndicated = [];
    isLoading = false;
    count = 0;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new LocationApiService(AppConfig.API_URL);

        this.loadLocations();
        this.loadSyndicatedLocations();
    }

    get locationsSortedByName() {
        return this.locations.sort((a, b) => {
            if (a.name < b.name) {
                return -1;
            }
            else if (a.name > b.name) {
                return 1;
            }
            return 0;
        });
    }

    loadLocations = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const locationsJson = yield this.apiService.fetchAllLocations(config);
            this.locations.replace(locationsJson.data.map((locationJson) => {
                return new Location(
                    this,
                    locationJson,
                );
            }));
            this.count = locationsJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`LocationStore failed to load locations. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadSyndicatedLocations = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const locationsJson = yield this.apiService.fetchSyndicatedLocations();

            this.locationsSyndicated.replace(locationsJson.map((locationJson) => {
                return new Location(
                    this,
                    locationJson,
                );
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`LocationStore failed to load syndicated locations. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateLocationFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const locationsJson = yield this.apiService.fetchLocationById(id);
        let location = this.locations.find((loc) => {
            return loc.id === id;
        });

        if (locationsJson.length === 0 && location) {
            // location no longer exists on the server, remove it locally
            yield this.removeLocation(location);
            location = null;
        }
        if (locationsJson.length > 0) {
            if (location) {
                location.updateFromJson(locationsJson[0]);
            }
            else {
                location = new Location(this, locationsJson[0]);
                this.locations.push(location);
            }
        }
        this.isLoading = false;
        return location;
    });

    removeLocation = flow(function* (location) { // eslint-disable-line func-names, require-yield
        this.locations.splice(this.locations.indexOf(location), 1);
    });

    addLocation = flow(function* (location) { // eslint-disable-line func-names, require-yield
        this.locations.push(location);
    });
}
decorate(LocationStore, {
    locations: observable,
    locationsSyndicated: observable,
    locationsSortedByName: computed,
    isLoading: observable,
});

export default LocationStore;
