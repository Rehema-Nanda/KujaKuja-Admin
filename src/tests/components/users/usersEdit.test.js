import React from "react";
import { mount } from "enzyme";
import bcryptjs from "bcryptjs";
import UsersEdit from "../../../components/users/edit";
import { StoreProvider } from "../../../StoreContext";
import CountryStore from "../../../stores/Country/CountryStore";
import Country from "../../../stores/Country/Country";
import LocationStore from "../../../stores/Location/LocationStore";
import Location from "../../../stores/Location/Location";
import UserStore from "../../../stores/User/UserStore";
import User from "../../../stores/User/User";

jest.mock("bcryptjs");
jest.mock("../../../stores/Country/CountryStore");
jest.mock("../../../stores/Location/LocationStore");
jest.mock("../../../stores/User/UserStore");

CountryStore.mockImplementation(() => {
    return {
        countries: [
            new Country(
                null,
                {
                    id: "1",
                    enabled: false,
                    name: "Afghanistan",
                    iso_two_letter_code: "AF",
                    geojson: null,
                    lat: "0.00000000",
                    lng: "0.00000000",
                    created_at: null,
                    updated_at: null,
                },
            ),
        ],
    };
});

const locations = [
    new Location(
        null,
        {
            id: "1",
            name: "Test Location",
            geojson: null,
            lat: "0.00000000",
            lng: "0.00000000",
            created_at: null,
            updated_at: null,
            country_id: "1",
            country_name: "Afghanistan",
        },
    ),
];

LocationStore.mockImplementation(() => {
    return {
        locations: locations,
        locationsSortedByName: locations,
    };
});

const users = [
    new User(
        null,
        {
            id: "1",
            email: "testuser@kujakuja.com",
            encrypted_password: "encrypted password",
            current_sign_in_at: "2020-01-01T00:00:00.000Z",
            created_at: null,
            updated_at: null,
            settlement_id: "1",
            settlement_name: "Test Location",
            is_admin: true,
            is_survey: false,
            is_service_provider: false,
        },
    ),
];

const mockUserDelete = jest.fn();
users[0].delete = mockUserDelete;

const mockUpdateUserFromServer = jest.fn(() => {
    return users[0];
});
UserStore.mockImplementation(() => {
    return {
        users: users,
        updateUserFromServer: mockUpdateUserFromServer,
        apiService: {
            isAuthenticated: jest.fn(),
            getProfile: jest.fn(),
        },
    };
});

describe("UsersEdit component", () => {
    const testProps = {
        history: {
            push: jest.fn(),
        },
        match: {
            params: {
                id: 1,
            },
        },
    };
    const wrapper = mount(<UsersEdit {...testProps} />, {
        wrappingComponent: StoreProvider,
    });

    it("should render correctly", () => {
        expect(mockUpdateUserFromServer).toHaveBeenCalled();
        wrapper.update();
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should push history when cancelForm function is called", () => {
        const e = { preventDefault: jest.fn() };
        const cancelForm = jest.spyOn(wrapper.instance(), "cancelForm");
        cancelForm(e);
        expect(testProps.history.push).toHaveBeenCalledWith("/users");
    });

    it("should change modal state when toggleDeleteConfirmationModal function is called", () => {
        const modalState = wrapper.instance().simpleState.isDeleteConfirmationModalOpen;
        const toggleDeleteConfirmationModal = jest.spyOn(wrapper.instance(), "toggleDeleteConfirmationModal");
        toggleDeleteConfirmationModal();
        expect(wrapper.instance().simpleState.isDeleteConfirmationModalOpen).toEqual(!modalState);
    });

    it("should call delete function on User object and push history when delete function is called", async () => {
        const deleteFn = jest.spyOn(wrapper.instance(), "delete");
        await deleteFn();
        expect(mockUserDelete).toHaveBeenCalled();
        expect(testProps.history.push).toHaveBeenCalledWith("/users");
    });

    it("should call bycrypt hash function on submit if passwords do not match", () => {
        const handleFormSubmit = jest.spyOn(wrapper.instance(), "handleFormSubmit");
        handleFormSubmit(null, { password: "new password" });
        expect(bcryptjs.hash).toHaveBeenCalled();
    });
});
