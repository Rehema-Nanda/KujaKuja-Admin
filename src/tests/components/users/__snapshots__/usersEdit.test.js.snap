// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersEdit component should render correctly 1`] = `
"<UsersEdit history={{...}} match={{...}}>
  <div className=\\"page\\">
    <div className=\\"title-background-parent\\">
      <div className=\\"title-background-shape\\" />
    </div>
    <Banner pageTitle=\\"team member\\" elementAction=\\"update\\" buttonFunction={[Function]} buttonDisabled={false}>
      <Container tag=\\"div\\">
        <div className=\\"container\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-sm-7 col-md-8\\">
                  <div className=\\"page-title\\">
                    <h1 className=\\"left\\">
                      Team member
                    </h1>
                    <p>
                      Manage your
                       
                      team member
                    </p>
                  </div>
                </div>
              </Col>
              <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-sm-5 col-md-4\\">
                  <div className=\\"page-title\\" />
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Container>
    </Banner>
    <ConfirmationModal buttonFunction={[Function]} toggle={[Function]} isOpen={false}>
      <Modal isOpen={false} toggle={[Function]} autoFocus={true} centered={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}} />
    </ConfirmationModal>
    <Container tag=\\"div\\">
      <div className=\\"container\\">
        <div className=\\"white-page-bg\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col lg={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-lg-8 offset-lg-2\\">
                  <div>
                    <AvForm onValidSubmit={[Function]} tag={[Function: Form]} model={{...}} validationEvent={{...}} method=\\"get\\" onSubmit={[Function: onSubmit]} onKeyDown={[Function: onKeyDown]} onInvalidSubmit={[Function: onInvalidSubmit]}>
                      <Form noValidate={true} action=\\"#\\" method=\\"get\\" onSubmit={[Function]} onKeyDown={[Function: onKeyDown]} className=\\"av-valid\\" tag=\\"form\\">
                        <form noValidate={true} action=\\"#\\" method=\\"get\\" onSubmit={[Function]} onKeyDown={[Function: onKeyDown]} className=\\"av-valid\\">
                          <Row tag=\\"div\\">
                            <div className=\\"row\\">
                              <Col tag=\\"div\\" widths={{...}}>
                                <div className=\\"col\\">
                                  <AvField type=\\"select\\" name=\\"settlement_id\\" label=\\"Location\\" required={true} value=\\"1\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"settlement_id\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"settlement_id\\" className=\\"\\">
                                              Location
                                            </label>
                                          </Label>
                                          <AvInput id=\\"settlement_id\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"select\\" name=\\"settlement_id\\" required={true} value=\\"1\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"select\\" name=\\"settlement_id\\" required={true} value=\\"1\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"settlement_id\\">
                                              <select size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"settlement_id\\" required={true} value=\\"1\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"settlement_id\\" className=\\"is-untouched is-pristine av-valid form-control\\">
                                                <option value=\\"\\">
                                                  Please select
                                                </option>
                                                <option value=\\"1\\">
                                                  Test Location
                                                </option>
                                              </select>
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"checkbox\\" name=\\"is_admin\\" label=\\"Admin Role\\" checked={true}>
                                    <AvGroup check={true} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={true} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-check\\">
                                          <AvInput id=\\"is_admin\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"checkbox\\" name=\\"is_admin\\" checked={true} validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"checkbox\\" name=\\"is_admin\\" checked={true} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} value={true} className=\\"is-untouched is-pristine av-valid\\" id=\\"is_admin\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"is_admin\\" checked={true} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} value={true} id=\\"is_admin\\" type=\\"checkbox\\" className=\\"is-untouched is-pristine av-valid form-check-input\\" />
                                            </Input>
                                          </AvInput>
                                          <Label for=\\"is_admin\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"is_admin\\" className=\\"\\">
                                              Admin Role
                                            </label>
                                          </Label>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"checkbox\\" name=\\"is_survey\\" label=\\"Survey Role\\" checked={false}>
                                    <AvGroup check={true} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={true} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-check\\">
                                          <AvInput id=\\"is_survey\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"checkbox\\" name=\\"is_survey\\" checked={false} validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"checkbox\\" name=\\"is_survey\\" checked={false} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} value={false} className=\\"is-untouched is-pristine av-valid\\" id=\\"is_survey\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"is_survey\\" checked={false} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} value={false} id=\\"is_survey\\" type=\\"checkbox\\" className=\\"is-untouched is-pristine av-valid form-check-input\\" />
                                            </Input>
                                          </AvInput>
                                          <Label for=\\"is_survey\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"is_survey\\" className=\\"\\">
                                              Survey Role
                                            </label>
                                          </Label>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"checkbox\\" name=\\"is_service_provider\\" label=\\"Service Provider Role\\" checked={false}>
                                    <AvGroup check={true} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={true} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-check\\">
                                          <AvInput id=\\"is_service_provider\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"checkbox\\" name=\\"is_service_provider\\" checked={false} validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"checkbox\\" name=\\"is_service_provider\\" checked={false} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} value={false} className=\\"is-untouched is-pristine av-valid\\" id=\\"is_service_provider\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"is_service_provider\\" checked={false} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} value={false} id=\\"is_service_provider\\" type=\\"checkbox\\" className=\\"is-untouched is-pristine av-valid form-check-input\\" />
                                            </Input>
                                          </AvInput>
                                          <Label for=\\"is_service_provider\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"is_service_provider\\" className=\\"\\">
                                              Service Provider Role
                                            </label>
                                          </Label>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"email\\" name=\\"email\\" label=\\"Email\\" required={true} validate={{...}} value=\\"testuser@kujakuja.com\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"email\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"email\\" className=\\"\\">
                                              Email
                                            </label>
                                          </Label>
                                          <AvInput id=\\"email\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"email\\" name=\\"email\\" required={true} validate={{...}} value=\\"testuser@kujakuja.com\\" validationEvent=\\"\\" trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"email\\" name=\\"email\\" required={true} value=\\"testuser@kujakuja.com\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"email\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"email\\" required={true} value=\\"testuser@kujakuja.com\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"email\\" type=\\"email\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"password\\" name=\\"password\\" label=\\"Password\\" required={true} value=\\"encrypted password\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"password\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"password\\" className=\\"\\">
                                              Password
                                            </label>
                                          </Label>
                                          <AvInput id=\\"password\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"password\\" name=\\"password\\" required={true} value=\\"encrypted password\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"password\\" name=\\"password\\" required={true} value=\\"encrypted password\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"password\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"password\\" required={true} value=\\"encrypted password\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"password\\" type=\\"password\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"password\\" name=\\"password_confirmation\\" label=\\"Password Confirmation\\" required={true} validate={{...}} value=\\"encrypted password\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"password_confirmation\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"password_confirmation\\" className=\\"\\">
                                              Password Confirmation
                                            </label>
                                          </Label>
                                          <AvInput id=\\"password_confirmation\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"password\\" name=\\"password_confirmation\\" required={true} validate={{...}} value=\\"encrypted password\\" validationEvent=\\"\\" trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"password\\" name=\\"password_confirmation\\" required={true} value=\\"encrypted password\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"password_confirmation\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"password_confirmation\\" required={true} value=\\"encrypted password\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"password_confirmation\\" type=\\"password\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                            </div>
                          </Row>
                          <BtnRow buttonLabel=\\"Update\\" cancelForm={[Function]}>
                            <Row className=\\"btn-row\\" tag=\\"div\\">
                              <div className=\\"btn-row row\\">
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\" />
                                </Col>
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\">
                                    <Button color=\\"primary\\" className=\\"primary cancel\\" onClick={[Function]} tag=\\"button\\">
                                      <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"primary cancel btn btn-primary\\" aria-label={{...}}>
                                        Back
                                      </button>
                                    </Button>
                                  </div>
                                </Col>
                              </div>
                            </Row>
                          </BtnRow>
                        </form>
                      </Form>
                    </AvForm>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </div>
    </Container>
  </div>
</UsersEdit>"
`;
